### PR TITLE
Fix bugzilla issue 24363 - hex string postfixes are useless

### PR DESCRIPTION
**Pull request by dkorpel**
`arrayCastBigEndian` is cut and pasted to utils.d with no further changes.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

**Comment by dkorpel** [comment]
CI fails on Windows with `Error: invalid UCS-32 char \Uaabbccdd`, I need to figure out where the hexstring gets UTF validated.

File: changelog/dmd.hexstring-cast.dd [hunk 1]
```diff
@@ -11,10 +11,19 @@ immutable uint[] data = cast(immutable uint[]) x"AABBCCDD";
...
 ---
 
-When the hex string has a character postfix, or its length is not a multiple of the element size, it is an error:
+Character postfixes can now also be used for integers of size 2 or 4:
+---
+immutable ushort[] f = x"80 3F"w;
+static assert(f[0] == 0x803F);
 
+immutable int[] f = x"80 35 FF FD"d;
+static assert(f[0] == 0x803FFF);
```
Comment (Lyratelle) [inline changelog/dmd.hexstring-cast.dd:20]: I don't understand the value in the assert. Shouldn't it be 0x8035FFFD ?!?

Comment (ibuclaw) [reply changelog/dmd.hexstring-cast.dd:20]: Looks like a copy-pase error from the above line @dkorpel

Comment (dkorpel) [reply changelog/dmd.hexstring-cast.dd:20]: Good dcatch

**Pull request merged**